From 2cce760b79d7e8e275f96e639f6002a5626ae4a6 Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Thu, 18 May 2017 14:42:12 +0800
Subject: [PATCH 5/7] arch/powerpc/platforrms/44x: acp_clk_release fix

the patch comes from:
http://git.yoctoproject.org/cgit/cgit.cgi/linux-yocto-3.14/commit/?h=standard/axxia/base&id=fa74a45f14a0a9c788239318874f83de1f6c68d5

This patch fixes a typo in acp_clk_release. The function was locking the same
mutex, instead of locking and unlocking the mutex, causing the kernel to hang.

Signed-off-by: Charlie Paul <cpaul.windriver@gmail.com>
Signed-off-by: Li Wang <li.wang@windriver.com>
---
 arch/powerpc/platforms/44x/acpclock.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/platforms/44x/acpclock.c b/arch/powerpc/platforms/44x/acpclock.c
index 71c13f8..281c097 100644
--- a/arch/powerpc/platforms/44x/acpclock.c
+++ b/arch/powerpc/platforms/44x/acpclock.c
@@ -129,7 +129,7 @@ static void acp_clk_release(struct kref *kref)
 
 	mutex_lock(&clocks_mutex);
 	list_del(&clk->node);
-	mutex_lock(&clocks_mutex);
+	mutex_unlock(&clocks_mutex);
 
 	kfree(clk);
 }
-- 
1.7.5.4

